### PR TITLE
Integrate pre-confirm dialog for remove gateway and credentials, Create visual page for edit gateway group membership

### PR DESCRIPTION
**Pull request by ajaypratap003**
### Type of change

- Enhancement / new feature
- Refactoring

### Description

- Intergrate pre confirm dialog for remove gateways and remove credentials

- Create generic GatewayGroupTypeAheadSelect component

- Add edit gatway group membership option in kebeb dropdown which will add based on the condition

- Create edit gateway group membership visual page

- Enhance error notification  in device detail page

- Enhance ActionManager component 

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: console/console-init/ui/src/constant/constants.ts [hunk 1]
```diff
@@ -82,6 +82,11 @@ enum Messages {
   InvalidJson = "Invalid json, please try again with valid json"
 }
 
+enum PageJourney {
```
Comment (drcoolsanjeev) [inline console/console-init/ui/src/constant/constants.ts:85]: please revisit this enum name.

File: console/console-init/ui/src/containers/GatewayGroupTypeAheadSelect/GatewayGroupTypeAheadSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+export const GatewayGroupTypeAheadSelect: React.FC<ITypeAheadSelectProps> = ({
+  id,
+  onSelect,
+  onClear,
+  selected,
+  placeholderText = "Input gateway group name",
+  isCreatable,
+  isMultiple,
+  "aria-label": ariaLebel = "gateway group dropdown",
```
Comment (drcoolsanjeev) [inline console/console-init/ui/src/containers/GatewayGroupTypeAheadSelect/GatewayGroupTypeAheadSelect.tsx:18]: typo in ariaLebel it should be `ariaLabel`

File: console/console-init/ui/src/modules/iot-device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx [hunk 3]
```diff
@@ -188,6 +198,18 @@ const DeviceDetailHeader: React.FunctionComponent<IDeviceDetailHeaderProps> = ({
         </DropdownItem>
       );
     }
+    if (memberOf?.length > 0) {
+      additionalKebabOptions.push(
+        <DropdownItem
+          id="device-detail-edit-credentials-dropdownitem"
+          key="edit-gateway-group-membership"
```
Comment (drcoolsanjeev) [inline console/console-init/ui/src/modules/iot-device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx:205]: please add unique key for each dropdown item

File: console/console-init/ui/src/modules/iot-device-detail/components/ConfigurationInfo/ConfigurationInfo.tsx [hunk 4]
```diff
@@ -114,26 +114,25 @@ export const ConfigurationInfo: React.FC<IConfigurationInfoProps> = ({
...
-        aria-label="Information needed to configure device"
-        position={TooltipPosition.bottom}
-        enableFlip={false}
-        content={
-          <>
-            This info section provides a quick view of the information needed to
-            configure a device connection on the device side.
-          </>
+      <Popover
+        position={PopoverPosition.bottom}
```
Comment (drcoolsanjeev) [inline console/console-init/ui/src/modules/iot-device-detail/components/ConfigurationInfo/ConfigurationInfo.tsx:118]: id and key are missing

File: console/console-init/ui/src/constant/constants.ts (hunk 1 above)
Comment (ajaypratap003) [reply console/console-init/ui/src/constant/constants.ts:85]: Purpose of this  enum is header change based on page so I thought to create enum as PageJourney which is indicate the page name. I feel this name is fine.

File: console/console-init/ui/src/modules/iot-device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx [hunk 5]
```diff
@@ -201,7 +201,7 @@ const DeviceDetailHeader: React.FunctionComponent<IDeviceDetailHeaderProps> = ({
     if (memberOf?.length > 0) {
       additionalKebabOptions.push(
         <DropdownItem
-          id="device-detail-edit-credentials-dropdownitem"
+          id="device-detail-edit-gateway-group-membership-dropdownitem"
           key="edit-gateway-group-membership"
```
Comment (drcoolsanjeev) [inline console/console-init/ui/src/modules/iot-device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx:205]: Please add a unique key

**Pull request merged**